test/org/zaproxy/zap/authentication/UsernamePasswordAuthenticationCredentialsUnitTest.java

```diff
@@ -0,0 +1,114 @@
+
+import org.zaproxy.zap.extension.api.ApiResponse;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+
+/**
+ * @author Vahid Rafiei (@vahid_r)
+ */
+public class UsernamePasswordAuthenticationCredentialsUnitTest {
+
+    private UsernamePasswordAuthenticationCredentials usernamePasswordAuthenticationCredentials;
+    private UsernamePasswordAuthenticationCredentials notConfiguredInstance;
+    private String username = "myUsername";
+    private String password = "myPassword";
+
+    @Before
+    public void setUp() {
+        this.usernamePasswordAuthenticationCredentials = new UsernamePasswordAuthenticationCredentials(username, password);
+        this.notConfiguredInstance = new UsernamePasswordAuthenticationCredentials();
+    }
+
+    @Test
+    public void shouldBeConfiguredIfUsernameAndPasswordAreNotNull() throws Exception {
+        // Given/When
+        boolean isConfigured = usernamePasswordAuthenticationCredentials.isConfigured();
+
+        // Then
+        assertThat(isConfigured, is(true));
+    }
+
+    @Test
+    public void shouldNotBeConfiguredIfUsernameAndPasswordAreNull() throws Exception {
+        // Given/When
+        boolean isConfigured = notConfiguredInstance.isConfigured();
+
+        // Then
+        assertThat(isConfigured, is(false));
+    }
+
+
+    @Test
+    public void shouldEncodeMethodReturnNullPatternIfUsernameIsNull() throws Exception {
+        // Given
+        String nullPattern = "AA==";
+        username = null;
+        password = "something";
+        String stringSeparator = "|";
+        usernamePasswordAuthenticationCredentials = new UsernamePasswordAuthenticationCredentials(username, password);
+
+        // When
+        String encodedResult = usernamePasswordAuthenticationCredentials.encode(stringSeparator);
+
+        // Then
+        assertThat(encodedResult, is(nullPattern));
+    }
+
+    @Test
+    public void shouldEncodeUsernameAndPasswordWithTheCorrectFieldSeparator() throws Exception {
+        // Given
+        List<String> someCorrectSeparators = Arrays.asList("-", "|", "/", "\\", "+");
+        String aCorrectSeparator = someCorrectSeparators.get(new Random().nextInt(someCorrectSeparators.size()));
+
+        // When
+        String encodedUsernamePassword = usernamePasswordAuthenticationCredentials.encode(aCorrectSeparator);
+
+        // Then
+        assertThat(encodedUsernamePassword, notNullValue());
+        assertThat(encodedUsernamePassword, is("bXlVc2VybmFtZQ==~bXlQYXNzd29yZA==~"));
+    }
+
+    @Test
+    public void shouldApiResponseRepresentationReturnApiResponseWithValidNameAndJsonFormat() throws Exception {
+        // Given/When
+        ApiResponse apiResponse = usernamePasswordAuthenticationCredentials.getApiResponseRepresentation();
+        JSON jsonRepresentation = apiResponse.toJSON();
+        String jsonResult= "{\"password\":\"myPassword\",\"type\":\"UsernamePasswordAuthenticationCredentials\",\"username\":\"myUsername\"}";
```

Comment:
Weird.. I'm getting the exact order and my tests pass. 
Maybe it's better to search for fields with `allOf()` matcher and check for the specific values..